packages/outstatic/src/utils/documentCommitInput.ts

```diff
@@ -3,7 +3,7 @@ import { FileType } from '../types'
 import { assertUnreachable } from './assertUnreachable'
 import { IMAGES_PATH } from './constants'
 
-type createCommitInputType = {
```

Comment:
We'll probably want to split these into a few functions. The Metadata DB also reuses the commit input generation so that we don't have duplication regarding how we send updates to GitHub.

A suggested refactor:

```ts
import {
  type FileChanges,
  type CreateCommitOnBranchInput,
} from "graphql/generated"

interface InputAPI {
  getInput: () => CreateCommitOnBranchInput
  setMessage: (title: string, body?: string) => void
  replaceFile: (file: string, contents: string) => void
  removeFile: (file: string) => void
}

export const createCommitInput = ({
  message,
  owner,
  oid,
  repoSlug,
  repoBranch,
}: InputAPI) => {
  const additions:FileChanges["additions"] = []
  const deletions: FileChanges["deletions"] = []
  let commitMessage = message ?? "chore: Outstatic commit"
  let commitBody = "Automatically created by Outstatic"

  const setMessage = (title: string, body?: string) => {
    commitMessage = title ?? commitMessage
    commitBody = body ?? commitBody
  }

  const replaceFile = (file: string, contents: string) => {
    additions.push({ path: file, contents })
  }

  const removeFile = (file: string) => {
    deletions.push({path: file})
  }

  const getInput = () => ({
    input: {
      branch: {
        repositoryNameWithOwner: `${owner}/${repoSlug}`,
        branchName: repoBranch,
      },
      message: {
        headline: commitMessage,
      },
      fileChanges: {
        additions: [],
        deletions: [],
      },
      expectedHeadOid: oid,
    },
  });

  // return the API
  return { setMessage, getInput, replaceFile, removeFile }
}
```

Given this API, you can then call helper functions such as `addDocumentToCommit(api, {/*options*/})` `updateCustomSchema(api, /*schema object*/)`, and then call `api.getInput()` to get a suitable GraphQL input for the mutation.